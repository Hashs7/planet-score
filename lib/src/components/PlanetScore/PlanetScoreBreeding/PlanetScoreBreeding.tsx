import { BREEDING_ANIMAL, BREEDING_ANIMAL_REGEX, BREEDING_CONDITION } from '@/constants';
import PlanetScoreBreedingAnimal from './PlanetScoreBreedingAnimal/PlanetScoreBreedingAnimal';
import PlanetScoreBreedingIcon from './PlanetScoreBreedingIcon/PlanetScoreBreedingIcon';
import PlanetScoreBreedingInfo from './PlanetScoreBreedingInfo/PlanetScoreBreedingInfo';

const PlanetScoreBreeding = ({ value }: { value: string }) => {
  const [animal] = (value.toLowerCase().match(BREEDING_ANIMAL_REGEX) as BREEDING_ANIMAL[]) || [];
  if (!animal) {
    return null;
  }

  const [, score] = value.toLowerCase().split(animal) as BREEDING_CONDITION[];

  return (
    <>
      <PlanetScoreBreedingInfo />
      <PlanetScoreBreedingAnimal animal={animal} score={score} />
      <PlanetScoreBreedingIcon score={score} />
    </>
  );
};

export default PlanetScoreBreeding;
