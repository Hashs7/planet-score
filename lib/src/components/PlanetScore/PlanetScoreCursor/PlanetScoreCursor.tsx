import { SCORE_POSITIONS, ScoreValues } from '@/constants';
import { FunctionComponent } from 'react';

interface PlanetScoreCursorProps {
  score: ScoreValues;
  translateY?: number;
}

const PlanetScoreCursor: FunctionComponent<PlanetScoreCursorProps> = ({
  score = '',
  translateY = 0
}) => {
  const index = SCORE_POSITIONS.indexOf(score.toUpperCase() as ScoreValues);
  const translateX = index * 108;

  if (index === -1) {
    return null;
  }

  return (
    <g className="cursor" transform={`translate(${translateX} ${translateY})`}>
      <path
        style={{ fill: '#FDFDFD' }}
        d="M572.2,346.6c-16,0-29,13-29,29s13,29,29,29c16,0,29-13,29-29S588.1,346.6,572.2,346.6z"
      />
      <path
        style={{ fill: '#273259' }}
        d="M572.2,336.6c-21.5,0-39,17.5-39,39s17.5,39,39,39c21.5,0,39-17.5,39-39S593.7,336.6,572.2,336.6z
		 M543.2,375.6c0-16,13-29,29-29c16,0,29,13,29,29s-13,29-29,29C556.2,404.6,543.2,391.6,543.2,375.6z"
      />
    </g>
  );
};

export default PlanetScoreCursor;
