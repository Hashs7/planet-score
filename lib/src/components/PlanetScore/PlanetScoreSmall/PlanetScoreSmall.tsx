import PlanetScoreSmallMainScore from './PlanetScoreSmallMainScore/PlanetScoreSmallMainScore';
import { ScoreValues } from '@/constants';

const PlanetScoreSmall = ({ score }: { score: ScoreValues }) => (
  <svg viewBox="0 0 470 586">
    <g id="planet-score-logo">
      <path
        style={{ fill: '#C7DDF3' }}
        d="M233,60.6c4.1-2.4,11.8,3.6,9.8-1.6s-8.6-9.1-9.3-10.8s-4.4-5-6.1-3.9c-3.3-2.9-3.6-2.8-5.7-2.8
			c-2.1,0-11.1,6.3-6.7,5.7c3.9-0.3,7.7-0.3,11.6,0c0,0.5,0,1,0,1.5c1.7,9.2,1.6,3.6,4.1,6.7C233.3,58.6,224.8,65.2,233,60.6z"
      />
      <path
        style={{ fill: '#C7DDF3' }}
        d="M214.6,64.4c3.6,0.5,7.9,0,8.8-3.1c0.9-3.1,5.2-5.4,0-5.7c-1.7-0.4-3.5-0.4-5.2,0.1
			c-1.7,0.5-3.3,1.3-4.6,2.5C210.5,60.8,214.6,64.4,214.6,64.4z"
      />
      <path
        style={{ fill: '#C7DDF3' }}
        d="M40.5,115.7l0.3-0.3c1.9-1.2,15.3-8.6,15.5-8.7c11.1-8.3,9-9.1,22.9-12.4
			c5.5-1.4,13.9-3.5,12.4,6.2c0,0.7,0.8,1,1.2,1.4c0.3,0.3,0.6,0.5,1,0.6c0.4,0.1,0.8,0.2,1.2,0.1c2.9-0.4,2.7,1.8,5.6-0.3
			c7.7-5.7,16.9-18.4,28.3-21.4s18.5,3.1,24.8,0c6.3-3.1,9.2-5.4,6.2-8.5c-3-3.1-5.3-4.6-0.6-7c6.8-3.4,10-3.7,11.3-2.8
			c0.6,0.3,1.2,0.6,1.9,0.6c0.7,0.1,1.4,0,2-0.2l4.5-1.4c0.7-0.2,1.4-0.6,2-1.1c0.5-0.6,0.9-1.2,1.1-2c0.2-0.8,0.1-1.5-0.1-2.3
			c-0.3-0.7-0.7-1.4-1.3-1.9c-0.4-0.4-0.7-1-0.8-1.6c-0.1-0.6,0-1.2,0.3-1.8l4.1-7.2c0.4-0.7,1.9-0.5,1.9-1.3c0-7.4,12.1-3.1,18-3.8
			c4.3,0,8.5-0.9,12.4-2.6c7.7-2.1,0.3-4.4,7.2-5.2s14-4.1,14-6.6s-20.7,1.4-28.4,0.9c-6.4,0.2-12.8,1.2-18.9,3.1
			c-0.2-0.2-0.4-0.4-0.7-0.5c-3.6-2.1-9.3-4.7-5.1-5.7c11.5-4.6,23.7-7.2,36-7.7c8.3,0,12-1.5,13.6-3.2c0.4-0.4,0.9-0.7,1.5-0.9
			c0.6-0.2,1.1-0.3,1.7-0.2c17.6,0.8,35.1,3.8,52,9c-0.2,0.5-0.4,0.9-0.6,1.4h-1.3C281.5,21,264,23,257.8,30.8
			c-6.2,7.7-11.3,1-10.8,8.8c0.5,7.7-1,7.7,4.1,8.3s13.5-2.1,12.4,1c-1.1,3.2,8.3,4.1-1.6,6.7c-9.8,2.6-2.6-3.6-9.8,2.6
			c-7.2,6.2-3.6,5.7-8.3,7.7c-4.6,2.1-14.4-0.5-15.4,3.1c-1,3.6,9.3,5.7,6.7,10.3c-2.6,4.6-8.3,3.1-15.5,4.6
			c-7.2,1.5-11.3,4.1-11.9,12.4c-0.4,7.2,11.9,5.8,14.5,8.3c0.5,0.6,1.2,1,2,1.2c5.4,0.4,10.9,0.4,12.4-2.8c3-7,0-12.9,7.7-13.9
			s5.7-1.6,9.3-4.6c3.6-3.1-0.5-3.1,8.3-2.1c8.8,1,22.7,7.7,20.7,10.3s-11.9,4.6-4.1,6.7c7.7,2.1,10.3,3.5,13.4-1
			c3.1-4.5,3.6-5.7,1-7.7c-2.6-2.1-12.4-7.7-7.2-6.7s16,8.8,22.7,19.1s0.5-6.2,5.2-7.2c4.6-1,7.7,10.8,16,11.3
			c8.2,0.5,14.4-0.5,16,3.6c1.5,4.1,6.2,15.5-2.1,11.8c-12.9-5.7-16.5,1-11.8,3.6l7.4,6.1c0.5,0.4,1.2,0.7,1.8,0.9s1.4,0.1,2,0
			c1-0.3,2-0.5,3.1-0.6l1.5-0.4l3.1-1.4c0.7-0.3,1.5-0.5,2.2-0.4c0.8,0.1,1.5,0.4,2.1,0.8c1.9,1.5,4.1,3.5,6.4,5.7
			c0.3,0.2,0.7,0.4,1.1,0.4c0.4,0,0.8-0.1,1.1-0.4c2.9-1.5,5.3-3.7,7.2-6.3c0.5-0.9,1.1-1.7,1.9-2.3c0.8-0.6,1.8-1,2.8-1.2l10.7-2.2
			c2.9-0.8,5.9-1.2,8.8-1.2h5.5c0.3,0,0.6,0,0.8-0.2c0.3-0.1,0.5-0.3,0.6-0.6l2.8-4.1c0.2-0.4,0.3-0.8,0.3-1.2
			c-0.3-1.2-0.2-2.4,0.3-3.5c1.1-0.4,2.2-0.6,3.4-0.7c1.7-2.1,3.2-4.1,4.8-5.7c-27-44.7-69.1-78.2-118.7-94.4
			C244.9-3,191.2-0.9,143,19.2l-2,0.9c-2.1,0.9-4.1,1.8-6.1,2.8l-3.5,1.6l-3.8,1.9c-3.1,1.6-6.1,3.2-9.1,5l-1.1,0.6
			C96.3,44.5,77.3,60.4,61.3,79.1l-0.5,0.5c-1.2,1.4-2.4,2.9-3.7,4.5l-2.6,3.3l-2.5,3.3c-1.8,2.5-3.6,5-5.3,7.5
			c-0.1,0.1-0.2,0.3-0.3,0.4c-4.5,6.3-8.4,12.9-11.8,19.8L40.5,115.7z M363.7,87.4c4.1,6.5,5.2,1.5,12.4,7.2
			c4.5,4.5,7.5,10.3,8.5,16.5c0.1,0.3,0,0.5,0,0.8c-0.1,0.3-0.2,0.5-0.4,0.7c-0.2,0.2-0.4,0.3-0.6,0.4c-0.2,0.1-0.5,0.1-0.8,0.1
			c-1.9-0.1-3.7-0.8-5.3-2c-1.5-1.2-2.6-2.8-3.3-4.6c-3-7.7-2.9-4.7-8.9-10.7c-3.3-3.3-6.1-7.1-8.3-11.2c-1.6-2.2-2.8-4.6-3.6-7.2
			C353.1,72.6,359.6,80.9,363.7,87.4z M320,75c6.2,0.5,5.7,3.6,11.8,3.6c6.2,0,13.9,5.7,17,8.7s-2.6,5.2-7.2,3.6
			c-4.6-1.5,0-3.6-4.6-1.6c-3.7,1.7-12,0-15.2-0.7c-0.4-0.1-0.7-0.3-0.9-0.6c-0.2-0.3-0.4-0.7-0.4-1C320.2,80.4,314.3,74.5,320,75z
			 M136.7,37.8c2.6-2.6,7.1-7.6,5.8-7.8c-11.1-0.8,13.3-6.2,18-7.5c4.7-1.4,11.9-2.4,11.1-1.1c-0.8,1.3-7.7,4.6-5.6,5.9
			c2.1,1.3-1.3,2.7-4.7,3.8c-3.9,0.6-7.8,1.5-11.5,2.7c-5.2,1.7-9.5,3.5-12.1,4.6c-0.6,0.4-1.4,0-1-0.5V37.8z"
      />
      <path
        style={{ fill: '#222D56' }}
        d="M111.5,240.1c2,0,4.4,0.3,7.1,0.8c2.7,0.5,5.1,1.4,7.1,2.6c2,1.2,3,2.8,3,4.7
				c0,1.7-0.5,3.1-1.6,4.3c-1.1,1.2-2.5,1.8-4.3,1.8c-0.8,0-1.7-0.1-2.6-0.4c-0.9-0.3-1.9-0.6-2.9-0.9s-2-0.6-3-0.8
				c-1-0.2-2.1-0.3-3.3-0.3c-2,0-3.7,0.5-5.1,1.4c-1.4,1-2.1,2.2-2.1,3.8c0,1.7,1.6,3.1,4.8,4.2c3.2,1,6.8,2.2,10.7,3.4
				s7.2,3,9.9,5.5c2.7,2.5,4.1,6.1,4.1,10.9c0,3.7-1,7-3.1,9.9c-2,2.9-4.8,5.1-8.2,6.7s-7.2,2.3-11.3,2.3c-5,0-9.9-0.7-14.5-2
				c-4.6-1.4-7-3.7-7-7.1c0-1.6,0.5-3,1.6-4.1c1.1-1.1,2.5-1.6,4.3-1.6c0.9,0,1.8,0.1,2.8,0.4c1,0.3,2.4,0.7,4,1.2
				c1.6,0.5,3,0.9,4.2,1.2c1.2,0.3,2.5,0.4,3.8,0.4c1.9,0,3.5-0.3,4.9-0.9c1.4-0.6,2.4-1.4,3.1-2.5c0.7-1.1,1.1-2.3,1.1-3.6
				c0-2.1-0.8-3.4-2.4-4c-1.6-0.6-5.2-1.8-10.8-3.4c-5.6-1.6-9.8-3.5-12.4-5.7s-3.9-5.3-3.9-9.6c0-3.7,1-6.9,3-9.7
				c2-2.8,4.7-4.9,8-6.4C104,240.9,107.6,240.1,111.5,240.1z"
      />
      <path
        style={{ fill: '#222D56' }}
        d="M140.2,270c0-4.2,0.7-8.1,2.2-11.8c1.5-3.6,3.6-6.8,6.3-9.6c2.7-2.7,5.9-4.8,9.6-6.3
				c3.7-1.5,7.8-2.2,12.3-2.2c1.8,0,3.8,0.2,6.1,0.5c2.2,0.3,4.3,0.8,6.2,1.5c1.9,0.7,3.5,1.6,4.8,2.8c1.3,1.2,1.9,2.6,1.9,4.4
				c0,1.5-0.5,2.8-1.6,3.8c-1.1,1.1-2.4,1.6-3.9,1.6c-0.9,0-2.3-0.3-4-0.9c-1.7-0.6-3.2-1.1-4.4-1.4s-3-0.5-5.1-0.5
				c-2.7,0-5.1,0.5-7.2,1.4c-2.1,1-3.8,2.3-5.2,3.9c-1.4,1.7-2.4,3.6-3.2,5.9c-0.7,2.2-1.1,4.6-1.1,7.2c0,5.1,1.5,9.4,4.4,12.8
				c2.9,3.4,7,5.2,12.2,5.2c2.2,0,3.9-0.2,5.2-0.6c1.3-0.4,2.8-1,4.7-1.7c1.8-0.7,3.2-1.1,4.2-1.1c1.8,0,3.2,0.6,4.3,1.9
				c1,1.3,1.6,2.5,1.6,3.6c0,2.4-1.1,4.3-3.3,5.7c-2.2,1.4-4.9,2.4-7.9,3c-3.1,0.6-5.9,0.9-8.7,0.9c-5.9,0-11.1-1.3-15.7-3.9
				s-8.2-6.2-10.8-10.8C141.5,280.6,140.2,275.5,140.2,270z"
      />
      <path
        style={{ fill: '#222D56' }}
        d="M228.2,299.9c-5.9,0-11.1-1.3-15.7-3.9s-8.2-6.2-10.8-10.8c-2.6-4.6-3.9-9.6-3.9-15.1
				c0-4.2,0.7-8.1,2.2-11.8c1.5-3.6,3.6-6.8,6.3-9.6c2.7-2.7,5.9-4.8,9.6-6.3c3.7-1.5,7.8-2.2,12.3-2.2c4.4,0,8.4,0.8,12.2,2.3
				c3.8,1.5,7,3.7,9.7,6.4c2.7,2.7,4.8,5.9,6.3,9.6c1.5,3.6,2.2,7.5,2.2,11.6c0,4.2-0.7,8-2.2,11.7c-1.4,3.6-3.5,6.8-6.3,9.6
				c-2.7,2.8-6,4.9-9.7,6.4C236.7,299.2,232.6,299.9,228.2,299.9z M228.2,288.7c2.7,0,5.1-0.5,7.2-1.5c2.1-1,3.9-2.3,5.3-4.1
				c1.4-1.8,2.5-3.8,3.1-6c0.7-2.2,1-4.6,1-7.1c0-3.4-0.6-6.5-1.9-9.3c-1.3-2.8-3.2-5.1-5.7-6.8c-2.5-1.7-5.5-2.6-9.1-2.6
				s-6.5,0.9-9.1,2.6c-2.5,1.7-4.4,4-5.7,6.8c-1.3,2.8-1.9,5.9-1.9,9.3c0,5.3,1.4,9.8,4.3,13.3S222.8,288.7,228.2,288.7z"
      />
      <path
        style={{ fill: '#222D56' }}
        d="M267.7,293.1v-45.6c0-2,0.6-3.5,1.7-4.7c1.2-1.1,2.9-1.7,5.1-1.7h15.5c7,0,12.5,1.5,16.3,4.5
				c3.8,3,5.7,7.6,5.7,13.9c0,2.2-0.4,4.2-1.3,5.9c-0.8,1.8-2,3.3-3.5,4.6s-3.2,2.3-5.2,3.1c-1.9,0.8-4.1,1.4-6.4,1.7l15.1,14.8
				c1.2,1.2,1.7,2.4,1.7,3.6c0,1.6-0.6,3.1-1.9,4.3c-1.3,1.3-2.8,1.9-4.7,1.9c-2.1,0-3.8-0.7-5.1-2.1l-19.6-21.9H281v17.6
				c0,2-0.6,3.6-1.8,4.7c-1.2,1.1-2.8,1.6-4.8,1.6c-1.9,0-3.5-0.6-4.8-1.7C268.3,296.6,267.7,295.1,267.7,293.1z M281,252.3v14.4
				h9.1c2.4,0,4.4-0.6,6-1.9c1.5-1.3,2.3-3,2.3-5.3c0-2.3-0.8-4.1-2.3-5.3s-3.5-1.9-6-1.9H281z"
      />
      <path
        style={{ fill: '#222D56' }}
        d="M320.5,292.8v-44.7c0-2.1,0.6-3.8,1.8-5.1c1.2-1.3,2.9-1.9,5.1-1.9h21.9c2,0,3.6,0.6,4.6,1.7
				c1.1,1.1,1.6,2.4,1.6,4c0,1-0.2,1.9-0.7,2.8c-0.5,0.8-1.2,1.5-2.1,2c-0.9,0.5-2.1,0.8-3.4,0.8h-15.6v11.8h14.5
				c2.1,0,3.7,0.5,4.8,1.6c1.1,1.1,1.7,2.4,1.7,4c0,1.5-0.6,2.8-1.7,3.9c-1.2,1.1-2.7,1.7-4.7,1.7h-14.5v12.3h16.4
				c2,0,3.6,0.5,4.7,1.6c1.1,1.1,1.6,2.4,1.6,4c0,0.9-0.2,1.8-0.7,2.7c-0.5,0.9-1.2,1.6-2.2,2.1c-0.9,0.5-2.1,0.8-3.4,0.8h-22.8
				c-2,0-3.6-0.6-4.9-1.7C321.1,296.1,320.5,294.6,320.5,292.8z"
      />
      <path
        style={{ fill: 'none' }}
        d="M39,145.8c-1.5,0.6-3.1,1.3-4.6,2c0,0-0.1,0-0.1,0.1c1.1,0.2,2.4,0.3,3.6,0.4
				C38.3,147.4,38.6,146.6,39,145.8z"
      />
      <path
        style={{ fill: '#222D56' }}
        d="M99.1,115.9c-1.2,0-2.4,0-3.7,0.1c-4.2,0.3-8.5,0.9-12.9,1.6c0.2-1.7,0.1-3.5-0.5-5.2
				c-0.6-1.6-1.6-3.1-2.9-4.3c-0.7-0.7-2.3-1-4.2-1c-8.4,0-24.7,5.3-21.8,8.1c1,1.2,1.8,2.7,2.2,4.2s0.4,3.2,0,4.8
				c-6.2,1.9-12.3,4.2-18.3,6.4c-8.8,3.2-17.2,7.3-25.2,12.3c-4.2,2.7-11.5,7.4-11.8,13.1c-0.5,8.8,14.4,10.9,20.6,11.2
				c1.2,0.1,2.4,0.1,3.6,0.1c4.8,0,9.7-0.4,14.5-1.2c-0.6,1.5-1.3,3-2,4.4c-8.5,18.9-15.9,38.1-20.8,51.7c-1.4,3.3-1.6,7-0.7,10.4
				c1.7,9.5,3.5,14.6,4.7,15.3c3.9-1.3,5.4-4.4,7.3-8.5C39.3,214.1,53,187.6,64.5,162c4.8-10.7,9.6-21.1,14.1-32
				c0.5-1.1,1-2.3,1.4-3.5c0.7,0,1.4-0.1,2.1-0.1c4.7,0,9.4,1,13.7,3.1c2.4,1.1,4.6,2.7,6.4,4.6c1.8,1.9,3.2,4.2,4.2,6.7
				c0.9,2.5,1.4,5.2,1.3,7.8c-0.1,2.7-0.7,5.3-1.8,7.7c-1.5,4.8-4.5,8.9-8.7,11.7c-3.6,2.4-7.9,3.7-12.2,3.7c-0.6,0-1.3,0-1.9-0.1
				c-5.3,0-6.6,3-10.9,6c-1.4,0.9-8.5,5.5-3,5.5c22.9,0,53.6-10.4,61.6-34.3C138.4,126.4,118.2,115.9,99.1,115.9z M51.1,138.1
				c-2.6,6.1-5.3,12.1-7.9,18.1c-1.8,0-3.6,0-5.4,0c-4.2-0.2-16.9-1.7-16.2-8.5c0.4-3.6,6.6-5.9,9.5-7.3c7.1-3.2,14.5-5.8,22-7.9
				C52.4,134.8,51.7,136.6,51.1,138.1z"
      />
      <path
        style={{ fill: '#FFFFFF' }}
        d="M21.6,147.7c-0.8,6.7,11.9,8.2,16.2,8.5c1.8,0,3.6,0,5.4,0c2.7-6,5.3-12,7.9-18.1
					c0.6-1.4,1.4-3.3,2-5.5c-7.5,2.1-14.9,4.7-22,7.9C28.2,141.8,22,144.1,21.6,147.7z M34.4,147.7c1.5-0.7,3-1.3,4.6-2
					c-0.4,0.8-0.7,1.6-1.1,2.4c-1.3-0.1-2.5-0.2-3.6-0.4C34.3,147.8,34.4,147.8,34.4,147.7z"
      />
      <path
        style={{ fill: '#222D56' }}
        d="M458.6,140l-10.6,0.4c-6.9,0-13.9,0.3-20.8,0.3c1.8-2.8,3.5-5.6,5.4-8.5l1.1-1.5
				c0.5-0.8,0.8-1.6,0.8-2.6c0-0.9-0.3-1.8-0.8-2.6l-7.7-10.4c-0.2-0.2-0.3-0.4-0.6-0.5c-0.2-0.1-0.5-0.2-0.7-0.3s-0.5,0-0.8,0.1
				c-0.2,0.1-0.5,0.2-0.7,0.4c-7.4,7.5-13.8,16-19.1,25.1c-4-0.4-7.8-0.8-14.7-1c-6.5,0.6-12.7,2.8-18.2,6.3c-1.5,1-7.2,5.9-2,6
				c11.5,0.2,18.3,0.3,27.7,0.4c-8.8,13.6-16.1,28.1-21.8,43.2c-1.3,0.7-2.5,1.6-3.6,2.5c-6.1,5.4-13.1,9.8-20.6,12.9
				c-8,3.2-21.9,3.1-25.7-4.8c13-1.1,25.4-6.1,35.5-14.3c5.5-4.8,12.3-14.7,5.7-21.1s-20.4-3.2-28.3,0c-15,5.8-33.1,19.6-35.1,37
				c-0.1,0.7-0.1,1.5,0,2.2c-4.4,3.6-11,7.5-13.1,2.4c-2.2-5.1,2-13,3.6-17.5c2.2-6.6,11.3-23.8-2.5-21.6
				c-14.7,2.4-26,14.5-36.2,24.1c-3.3,3.1-6.7,6.3-10.1,9.4c1.3-11.1,10.4-23.2,16.6-32.2c2.7-4.2-3.4-3.2-5.7-2.7
				c-5.4,0.9-10.4,3.4-14.2,7.3c-6.8,8.9-12.3,18.6-16.5,28.9c-3,2.3-7.2,5.5-9.2,5.1c-3.8-0.9-4.2-5.8-4.2-9
				c0.6-6.2,2.2-12.4,4.8-18.1c1.6-4.2,5.9-10.3,1.8-14.4c-8.7-8.5-31.1-0.5-39.4,3.4c-9.7,4.1-17.7,11.5-22.4,20.9
				c-1.1,2.4-2.1,4.8-2.9,7.3c-0.6,1.9-1,3.9-1.1,5.9c-4.2,3.3-9,6-14,7.9c-1.2-0.1-2.3-0.5-3.2-1.4c-2.8-2.5-2.4-8.1-2.2-11.4
				c0.3-5.7,1.3-11.4,3-16.9c28.4-15.2,51.2-39,65.3-67.9c3.3-6.1,4.2-14.5-4.9-15.1c-19-1.2-38.7,15.3-50.7,28.8
				c-10.9,12.7-19.6,27.1-25.6,42.7c-1,2.5-1.9,5-2.7,7.5c-4.1,1.6-8.3,2.8-12.7,3.6c-0.4,0.3-9.8,9.9-7.9,10.4h0.3
				c6-0.8,11.8-2.2,17.5-4.2c-1.2,4.6-2.2,9.3-3,14c-0.9,5.6-2.4,13.6,1.4,18.3c7.7,9.6,27-0.6,35-5l6-3.5c0.4,0.7,1,1.3,1.6,1.9
				c3.6,2.2,7.9,3,12,2.1c7.8-0.9,15.5-3.1,22.6-6.6c0,2.6,1.1,5.2,2.9,7.1c4.2,4.2,11.9,2.6,17.1,1.3c4.6-1.3,9.1-3,13.4-5.2
				c1.5,11.8,24.8-1.4,30.7-6c6.8-5.3,13-11.4,19.3-17.3l2.5-2.2c-0.7,2.1-1.5,4.2-2.2,6.2c-2.2,6.4-5.7,13.8-5.1,20.8
				c0.8,9.8,14.5,6.4,20.9,4.8c5.6-1.5,11.1-3.6,16.3-6.3c6.8,8.9,26.3,6.8,35.9,4.2c11.3-3.2,22.1-8,32-14.2
				c0,9.3,4.2,17.9,15.5,19c16.2,1.6,35.1-6.5,48.7-15.8c1.2-0.8-3.5-6.2-5.4-5.5c-3.1,1.1-5.9,2.8-8.5,4.9
				c-6.7,5.1-20.8,9.2-26.4,0.7c-4.8-7.3,1.8-16.9,5.3-25c4.1-9.4,8.7-18.6,13.9-27.4c1.3-2.2,2.7-4.4,4.2-6.6
				c13.7-1.1,27.2-3.2,40.6-6.1c1.4-0.3,2.3,0.5,8.2-4.6C471.2,139,464.3,139.6,458.6,140z M346.1,180.3c3.8,4.2-0.4,12.4-3.9,13.8
				c-5,1.3-10.2,1.7-15.3,1c-0.4,0-0.8,0-1.2,0C328.9,186.3,338.5,171.4,346.1,180.3z M157,139.8c5.7-8.3,12.5-15.7,20.3-22
				c0,0,4.2-3.4,5-1.1c0.4,1.7,0.2,3.6-0.5,5.2c-4.6,12.7-15.8,25.7-24.6,35c-4,4.2-12.3,11.8-16.9,15.5
				C144.8,161,150.4,150.1,157,139.8L157,139.8z M196.1,188.8c-1.8,4.2-3.3,8.5-4.5,12.9c-0.3,0.9-0.5,1.9-0.8,2.9l-2.3,1.6
				c-3.7,2.7-8.9,4.2-11.3,3c-4.9-3-3.9-8.7-2.5-13.6c1.2-4.9,3.4-9.5,6.4-13.6c0.8-0.9,1.8-1.6,2.9-2c1.1-0.5,2.3-0.7,3.5-0.7h3.7
				c1.5-0.1,2.9,0.3,4.2,0.9C198.3,181.9,198,184.1,196.1,188.8L196.1,188.8z"
      />
    </g>
    <PlanetScoreSmallMainScore score={score} />
  </svg>
);

export default PlanetScoreSmall;
