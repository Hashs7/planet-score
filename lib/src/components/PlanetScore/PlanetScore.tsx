import { FunctionComponent } from 'react';
import PlanetScoreBreeding from './PlanetScoreBreeding/PlanetScoreBreeding';
import PlanetScoreCursor from './PlanetScoreCursor/PlanetScoreCursor';
import PlanetScoreInfo from './PlanetScoreInfo/PlanetScoreInfo';
import PlanetScoreMainScore from './PlanetScoreMainScore/PlanetScoreMainScore';
import PlanetScoreSmall from './PlanetScoreSmall/PlanetScoreSmall';
import { ScoreValues } from '@/constants';

interface PlanetScoreProps {
  value: string;
  small?: boolean;
  className?: string;
}

const PlanetScore: FunctionComponent<PlanetScoreProps> = ({ value, small, className = '' }) => {
  if (!value) {
    return null;
  }

  const [rating, breeding] = value.split('-');
  const [score, pesticide, biodiversity, climate] = rating.toUpperCase().split('') as ScoreValues[];

  if (small) {
    return <PlanetScoreSmall score={score} />;
  }

  return (
    <svg viewBox="0 0 1418 622" className={className}>
      <PlanetScoreInfo />
      <PlanetScoreMainScore score={score} />
      <PlanetScoreCursor score={pesticide} />
      <PlanetScoreCursor score={biodiversity} translateY={85} />
      <PlanetScoreCursor score={climate} translateY={170} />
      {breeding ? <PlanetScoreBreeding value={breeding} /> : null}
    </svg>
  );
};

export default PlanetScore;
