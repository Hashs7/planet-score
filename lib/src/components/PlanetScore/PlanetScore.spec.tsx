import { render } from '@testing-library/react';
import PlanetScore from './PlanetScore';
import PlanetScoreBreeding from './PlanetScoreBreeding/PlanetScoreBreeding';
import PlanetScoreCursor from './PlanetScoreCursor/PlanetScoreCursor';
import PlanetScoreInfo from './PlanetScoreInfo/PlanetScoreInfo';
import PlanetScoreMainScore from './PlanetScoreMainScore/PlanetScoreMainScore';
import PlanetScoreSmall from './PlanetScoreSmall/PlanetScoreSmall';

jest.mock(`./PlanetScoreBreeding/PlanetScoreBreeding`);
jest.mock(`./PlanetScoreCursor/PlanetScoreCursor`);
jest.mock(`./PlanetScoreInfo/PlanetScoreInfo`);
jest.mock(`./PlanetScoreMainScore/PlanetScoreMainScore`);
jest.mock(`./PlanetScoreSmall/PlanetScoreSmall`);

const context = {};

describe('<PlanetScore/>', () => {
  it('should render null with invalid value', () => {
    const { container } = render(<PlanetScore value="" />);

    expect(container).toMatchSnapshot();
  });

  it('call all sub-components with each associated props', () => {
    render(<PlanetScore value="ABCD-bovinV" />);

    expect(PlanetScoreSmall).not.toHaveBeenCalled();
    expect(PlanetScoreInfo).toHaveBeenCalled();
    expect(PlanetScoreMainScore).toHaveBeenCalledWith(
      {
        score: 'A'
      },
      context
    );
    expect(PlanetScoreCursor).toHaveBeenCalledWith(
      {
        score: 'B'
      },
      context
    );
    expect(PlanetScoreCursor).toHaveBeenCalledWith(
      {
        score: 'C',
        translateY: 85
      },
      context
    );
    expect(PlanetScoreCursor).toHaveBeenCalledWith(
      {
        score: 'D',
        translateY: 170
      },
      context
    );
    expect(PlanetScoreBreeding).toHaveBeenCalledWith(
      {
        value: 'bovinV'
      },
      context
    );
  });

  it("prevent call PlanetScoreBreeding when score value doesn't contain animal", () => {
    render(<PlanetScore value="ABCD" />);

    expect(PlanetScoreBreeding).not.toHaveBeenCalled();
  });

  it('(small) call only <PlanetScoreSmall/> with score prop', () => {
    render(<PlanetScore value="ABCD-pouleV" small />);

    expect(PlanetScoreSmall).toHaveBeenCalledWith(
      {
        score: 'A'
      },
      context
    );
    expect(PlanetScoreBreeding).not.toHaveBeenCalled();
    expect(PlanetScoreInfo).not.toHaveBeenCalled();
    expect(PlanetScoreMainScore).not.toHaveBeenCalled();
    expect(PlanetScoreCursor).not.toHaveBeenCalled();
  });
});
