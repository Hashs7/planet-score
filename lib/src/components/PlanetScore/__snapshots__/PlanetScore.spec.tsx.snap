// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlanetScore/> should render null with invalid value 1`] = `<div />`;
